### DISCUSSION TITLE
Artifacts UI blank error:  Failed to ensure the relay has a Service Worker registered. Service Worker API is not supported in this browser

### DISCUSSION THREAD

**Discussion by roman-doubrava** [danny-avila/LibreChat#4609] [Troubleshooting]
### What happened?

Dear LibreChat,

when using the Artefact UI we get a blank screen in the Preview Mode with the following error:

__csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:439  Failed to ensure the relay has a Service Worker registered. See details below.
(anonymous) @ __csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:439
__csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:442  Invariant Violation: Failed to start the relay Service Worker: Service Worker API is not supported in this browser at https://266c9f6c163ac60-preview.sandpack-static-server.codesandbox.io/__csb_relay/__csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:456:3 at 
https://266c9f6c163ac60-preview.sandpack-static-server.codesandbox.io/__csb_relay/__csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:457:3
(anonymous) @ __csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:442
TypeError: Cannot read properties of undefined (reading 'ready')  at start (__csb_relay.k6zqcaldwongir8b9mnqriffbwpbwcr.js:445:35)

We tried also to set "chrome://flags/#unsafely-treat-insecure-origin-as-secure" to that host "266c9f6c163ac60-preview.sandpack-static-server.codesandbox.io" but it did not resolve the issue.

We tested with Chrome(Version 122.0.6261.129 (Official Build) (64-bit)) and Edge (130.0.2849.56 (Official build) (64-bit))

Best regards

### Steps to Reproduce

1. Install Librechat 
2. Settings -> Toggle Artifacts UI -> Enable
3. Generate Artefact with Prompt
4. Preview Artefact

### Screenshots

![image](https://github.com/user-attachments/assets/a60f426a-e68b-4f81-b6ca-74deb2de792c)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Works on both Chrome & Edge, does your network block any traffic? Artifacts works by pulling dependencies from codesandbox.io

**Comment by roman-doubrava** [comment]
We found a workarround putting our hostname into: chrome://flags/#unsafely-treat-insecure-origin-as-secure works after relaunch

**Comment by quacrobat** [comment]
Having same issue, tried putting the hostname (https://librechat:3080) in /flags/#unsafely... and enabling the flag. Doesn't work, preview is still blank. code is fine. 
![image](https://github.com/user-attachments/assets/a9bfa882-7613-4454-97c2-1d007b24a3f2)
![image](https://github.com/user-attachments/assets/70e35b5a-158e-40d8-98fe-75f97bcaed47)
 
![image](https://github.com/user-attachments/assets/5ff38fe6-06ef-444c-b0b3-823be94800dc)

**Reply by quacrobat** [reply]
Apologies for the quick post, but it's resolved, it should be http://librechat:3080 in our case, not https.
Hope this helps someone else.
Kudos @danny-avila and contributors for a great product!
